xtokens/src/lib.rs

```diff
@@ -763,16 +763,19 @@ pub mod module {
 		fn buy_execution(
 			asset: MultiAsset,
 			at: &MultiLocation,
-			weight: Weight,
+			weight: Option<Weight>,
 		) -> Result<Instruction<()>, DispatchError> {
 			let ancestry = T::LocationInverter::ancestry();
 			let fees = asset
 				.reanchored(at, &ancestry)
 				.map_err(|_| Error::<T>::CannotReanchor)?;
-			Ok(BuyExecution {
-				fees,
-				weight_limit: WeightLimit::Limited(weight),
-			})
+			let weight_limit = if let Some(limit) = weight {
+				WeightLimit::Limited(limit)
+			} else {
+				WeightLimit::Unlimited
+			};
```

Comment:
`nit` but we could make this a little more succint and elegant imo:

```suggestion
			let weight_limit = limit
			  .map(|x| WeightLimit::Limited(x))
			  .unwrap_or(WeightLimit::Unlimited);
```